### PR TITLE
Navbar responsiveness

### PR DESCRIPTION
**Pull request by vlasebian**
This PR implements screen size responsiveness for the navbar and for the footer. When the screen size is small, a side navbar is shown. I added a layout for the whole application and also did some several fixes.The side navbar is not finished.

---
This change is [Reviewable](https://reviewable.io/reviews/florinrusu96/best_site_frontend/22)

### CODE REVIEW

**Comment by vlasebian** [comment]
Before I opened the PR, I did not pull the latest changes from master branch. The last commit fixed this.

File: frontend/src/app/app.component.css [hunk 1]
```diff
@@ -3,4 +3,16 @@
...
     background: white;
     box-shadow: 1px 0px 11px 0px rgba(0, 0, 0, 0.70);
+}
+
+mat-sidenav-container, mat-sidenav-content, mat-sidenav {
+    height: 100%;
+}
+ 
+mat-sidenav {
+    width: 250px;
```
Comment (florinrusu96) [inline frontend/src/app/app.component.css:13]: we should avoid using pixels, use em/rem/vw/vh
https://medium.com/@julienetienne/pixels-are-dead-faa87cd8c8b9

File: frontend/src/app/footer/footer.component.css [hunk 2]
```diff
@@ -20,53 +23,47 @@
...
-
-.footer-best-bucharest-text{
-    top:55%;
-    text-transform: uppercase;
-    font-weight: bold;
-}
-
-.footer-university-text{
-    top:65%;
+    font-size: 20px;
```
Comment (florinrusu96) [inline frontend/src/app/footer/footer.component.css:26]: same, avoid using px, em or rem

File: frontend/src/app/home-page/home-page.component.ts [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+
 @Component({
   selector: 'app-home-page',
   templateUrl: './home-page.component.html',
   styleUrls: ['./home-page.component.css']
 })
 export class HomePageComponent implements OnInit {
+  /* Breakpoint is the number of columns that is shown. */
+  breakpoint: number = 2;
+  maxWidth: number = 600;
```
Comment (florinrusu96) [inline frontend/src/app/home-page/home-page.component.ts:22]: make it public, if you're using it in the html, or private if not

File: frontend/src/app/navigation/navigation.component.css [hunk 4]
```diff
@@ -1,17 +1,63 @@
+mat-toolbar {
+    border-radius: 3px;
```
Comment (florinrusu96) [inline frontend/src/app/navigation/navigation.component.css:2]: this is ok with px 👍

File: frontend/src/app/navigation/navigation.component.css [hunk 5]
```diff
@@ -1,17 +1,63 @@
...
+}
+
+span.navbar_link {
+    font-size: 20px;
+}
+
+.navbar_link:hover {
+    border-bottom: 3px solid #ffa615;
+    border-radius: 1px;
+    padding-top: 3px;
```
Comment (florinrusu96) [inline frontend/src/app/navigation/navigation.component.css:34]: paddings and margins, should be ok with px, but em/rm are still better imo

File: frontend/src/app/home-page/home-page.component.ts (hunk 3 above)
Comment (vlasebian) [reply frontend/src/app/home-page/home-page.component.ts:22]: This will disappear in the next pr

**Pull request merged**